app/src/marketplace/containers/EditProductPage/EditControllerProvider.jsx

```diff
@@ -91,6 +95,22 @@ function useEditController(product: Product) {
         }
     }, [isAnyTouched])
 
+    useEffect(() => {
```

Comment:
Not sure if this is the best place to load these.. this could be move to `EditProductPage/index.jsx` where we load some other stuff as well (line 61):

```js
    useEffect(() => {
        loadEthIdentities()
        loadDataUnion(beneficiaryAddress)
    }, [beneficiaryAddress, loadDataUnion, loadEthIdentities])
```

Looking at this file now, I noticed that it seems to have the unload handler code block defined twice:

```js
useEffect(() => {
        const handleBeforeunload = (event) => {
            if (isAnyTouched()) {
                const confirmationMessage = 'You have unsaved changes'
                const evt = (event || window.event)
                evt.returnValue = confirmationMessage // Gecko + IE
                return confirmationMessage // Webkit, Safari, Chrome etc.
            }
            return ''
        }

        window.addEventListener('beforeunload', handleBeforeunload)

        return () => {
            window.removeEventListener('beforeunload', handleBeforeunload)
        }
    }, [isAnyTouched])
```

Reply:
Yeah, I'll remove the duplicate block 👍

We could move stats loading there. Then we could propagate stats as a property for `EditControllerProvider`. Or did you have something else in mind?

Reply:
I think it doesn't need to be passed, they come from the store anyway and can be accessed with selectors. I was mainly thinking about putting the fetching methods in one place.

Reply:
Ah, right. I didn't notice they are stored in Redux. I'll fix 👍

Reply:
Now it should be good to go :)